keps/sig-api-machinery/1027-api-unions/README.md

```diff
@@ -86,42 +120,133 @@ become possible.
 
 ### Goals
 
-The goal is to enable a union or "oneof" semantics in Kubernetes types, both for
-in-tree types and for CRDs.
+Provide a simple mechanism for API authors to label fields of a resource as
+members of a oneOf, in order to receive standardized:
+
+* Validation - ensuring only one member field is set (or at most one if
```

Comment:
Specifically, I want people to *not have to write such validation*, reducing webhook usage & built-in validation code, ideally.